network.go

```diff
@@ -720,6 +720,15 @@ func (n *network) CreateEndpoint(name string, options ...EndpointOption) (Endpoi
 			ep.iface.mac = mac
 		}
 	}
+	// This may change depending how Aliases are transmitted by daemon
```

Comment:
Thanks John.

I like it that libnetwork is in control of what is being passed to the IPAM
driver, and that that is controlled via capabilities.

The only concern is that we introduced the separation between IPAM/Network
plumbing, and we just need to make sure that these new capabilities cannot
be used to couple the twos back together, fro ma remote driver which
implements both network and ipam, for example.

I think this is good to publish. So all maintainers can see it.

Thanks,
Alessandro

P.S.
Why not "Requires[...]"  instead of "Wants[...]" for the properties name ?
Maybe it was wrong in first place to call the first one
"RequiresMACAddress", but now I'd suggest to keep the names consistent.

On Fri, Feb 26, 2016 at 1:13 PM, JC Martin <notifications@github.com> wrote:

> In network.go
> <https://github.com/docker/libnetwork/pull/976#discussion_r54302842>:
>
> > @@ -720,6 +720,15 @@ func (n *network) CreateEndpoint(name string, options ...EndpointOption) (Endpoi
> >  			ep.iface.mac = mac
> >  		}
> >  	}
> > +	// This may change depending how Aliases are transmitted by daemon
>
> If we do, the alias will be specific to the each endpoint object (by the
> way - there is already aliases and myAliases - which will cause potential
> confusion with people used to linux terminology).
> In common implementation, IP aliases are assigned to loopback, and not a
> specific interface. it doesn't seem like this is possible from a plugin.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/libnetwork/pull/976/files#r54302842>.
>